app/src/main/java/org/projectbuendia/client/ui/SnackBar.java

```diff
@@ -0,0 +1,436 @@
+// Copyright 2015 The Project Buendia Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+// OR CONDITIONS OF ANY KIND, either express or implied.  See the License for
+// specific language governing permissions and limitations under the License.
+
+package org.projectbuendia.client.ui;
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.os.CountDownTimer;
+import android.support.annotation.NonNull;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.animation.TranslateAnimation;
+import android.widget.ExpandableListView;
+import android.widget.ImageView;
+import android.widget.SimpleExpandableListAdapter;
+import android.widget.TextView;
+
+import org.projectbuendia.client.R;
+import org.projectbuendia.client.utils.Utils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Snackbars provide lightweight feedback about an operation by showing a brief message at the
+ * bottom of the screen. Snackbars can contain an action button with an OnClickListener.
+ * Can be auto dismissed by a time out values in seconds and can be prioritized.
+ */
+public class SnackBar {
+
+    private final ViewGroup mTargetParent;
+    private final Context mContext;
+    private ExpandableListView mList;
+    private SnackBarListAdapter adapter;
+    private ArrayList<HashMap<String, String>> groups;
+    private ArrayList<ArrayList<HashMap<String, String>>> children;
+    private int mMessageId;
+    private TreeMap<MessageKey, Message> mMessagesList;
+
+
+    public SnackBar(ViewGroup parent) {
+        mTargetParent = parent;
+        mContext = parent.getContext();
+
+        mMessagesList= new TreeMap<>();
+        children = new ArrayList<>();
+        buildList();
+    }
+
+    /**
+     * Show the SnackBar with an slide up animation if hidden.
+     */
+    public void show() {
+        // TODO: discover why does not animate on the first time
+        if (mList.getVisibility() != View.VISIBLE) {
+            animate(View.VISIBLE);
+        }
+    }
+
+    /**
+     * Hide the SnackBar with a slide down animation if visible.
+     */
+    public void hide() {
+        animate(View.GONE);
+    }
+
+    /**
+     * Play the slide up / down animation depending on the list current visibility.
+     * @param visibility Defines which of the animations will be executed (up or down). Possible
+     *                   values are {@code View.VISIBLE} or {@code View.GONE}
```

Comment:
You can use `{@link View#VISIBLE}` for these instead of `{@code}`, which makes it possible to navigate to them directly in your IDE.